compile-all.sh

```
@@ -0,0 +1,121 @@
+# Usage: ./compile-all.sh [debug|release]   (default: debug)
+#
+# Examples are built sequentially so each can be timed on its own. cargo caches
+# the shared workspace dependencies, so only the FIRST build pays to compile
+# them — the remaining timings reflect each example's own crate. (One combined
+# `cargo build -p ... -p ...` would be marginally faster but can't time each.)
+#
+# Portable across Linux (GNU) and macOS (bash 3.2 + BSD find/awk/date).
+
+set -uo pipefail
+
+cd "$(dirname "$0")"
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)